### PR TITLE
Add Bedrock + Sagemaker support

### PR DESCRIPTION
**Pull request by billytrend-cohere**
This PR adds AWS integration to the Cohere client. 

- It adds a new `src/aws-utils.ts` module that provides utility functions for working with AWS.
- New `AwsClient`, `BedrockClient`, and `SagemakerClient` classes are added in `src/AwsClient.ts`, `src/BedrockClient.ts`, and `src/SagemakerClient.ts` respectively. 
- The `package.json` file is updated to include new dependencies required for AWS integration, such as `@aws-sdk/client-sagemaker`, `@aws-sdk/credential-providers`, `@aws-sdk/protocol-http`, and `@aws-sdk/signature-v4`.
- The `test.yml` file is updated to include AWS-related environment variables.
- The `src/index.ts` file is modified to export the new `BedrockClient` and `SagemakerClient` classes.
- New test files `src/test/aws-util-tests.test.ts` and `src/test/bedrock-tests.test.ts` are added to test the AWS integration.

### CODE REVIEW

**Comment by billytrend-cohere** [comment]
waiting on fern feature to make the tests pass but have verified that they pass locally

File: src/BedrockClient.ts [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        breadcrumbsPrefix: ["response"],
+    }
+
+    const parsed = await parser.parseOrThrow(obj, config)
+    return parser.jsonOrThrow(parsed, config);
+}
+
+export class BedrockClient extends CohereClient {
+    constructor(protected readonly _options: CohereClient.Options & BedrockProps) {
+        _options.token = "n/a";
```
Comment (mkozakov) [inline src/BedrockClient.ts:58]: should this be `null`?

File: src/BedrockClient.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            region: props.awsRegion,
+            credentials,
+            sha256: Sha256,
+        });
+
+        const headers =
+            !req.headers ? {}
+                : Array.isArray(req.headers) ?
+                    Object.fromEntries(Array.from(req.headers).map((header) => [...header]))
+                    : { ...req.headers };
```
Comment (mkozakov) [inline src/BedrockClient.ts:110]: nit: readability

```suggestion
let headers = {}
if (req.headers) {
  if (Array.isArray(req.headers)) {
    // construct headers object from array elements
    headers = Object.fromEntries(Array.from(req.headers).map((header) => [...header]))
  } else {
    headers = { ...req.headers}
  }
} 
```

File: src/BedrockClient.ts [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            return "embed";
+        }
+        if (generateModel && url.includes(generateModel)) {
+            return "generate";
+        }
+        throw new Error(`Unknown endpoint in url: ${url}`);
+    }
+
+    private parseAWSEvent(line: string) {
+        const regex = /{[^\}]*}/;
```
Comment (mkozakov) [inline src/BedrockClient.ts:144]: this might need some testing

**Review by mkozakov — APPROVED** [review]

File: src/BedrockClient.ts (hunk 1 above)
Comment (billytrend-cohere) [reply src/BedrockClient.ts:47]: Added comment

File: src/BedrockClient.ts (hunk 2 above)
Comment (billytrend-cohere) [reply src/BedrockClient.ts]: fixed

**Pull request merged**